itests/raft_messagesigner_test.go

```diff
@@ -507,27 +499,33 @@ func TestGoRPCAuth(t *testing.T) {
 	pkey2, _ := generatePrivKey()
 
 	pkeys := []*kit.Libp2p{pkey0, pkey1, pkey2}
```

Comment:
How was the third node able to become a leader when it wasn't part of the init peerset?

Reply:
If a node isn't part of a cluster it shouldn't be able to connect to it, and definitely shouldn't be able to become a leader - arguably this test was failing correctly, showing that it's possible for external nodes to "break" into a cluster

Reply:
(note that raft/gorpc are on the public libp2p interface, and being in a cluster is the only thing preventing calls to MpoolPushMessage by libp2p peers; If that check can be defeated that would be extremely bad)

Reply:
Okay, let's parking lot this for standup